checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,285 @@
+
+    // Creates a file to save user-facing messages
+    File reportFile = new File("./MapReduceIntegrationReport.txt");
+    try (PrintWriter reportWriter = new PrintWriter(reportFile)) {
+      SimpleDateFormat df = new SimpleDateFormat("dd/MM/yy HH:mm:ss");
+      Date date = new Date();
+      reportWriter.printf("%n%n%n***** The integration checker ran at %s. *****%n%n",
+          df.format(date));
+
+      if (!cannotFindClassAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "cannot recognize Alluxio classes.%n%n", cannotFindClassAddresses);
+      }
+      if (!cannotFindFSAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "cannot recognize Alluxio filesystem.%n%n", cannotFindFSAddresses);
+      }
+      if (!successAddresses.equals("")) {
+        reportWriter.printf("Hadoop nodes of IP addresses: %s "
+            + "can recognize Alluxio filesystem.%n%n", successAddresses);
+      }
+
+      if (!canFindClass) {
+        reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+        reportWriter.println(TEST_FAILED_MESSAGE);
+      } else if (!canFindFS) {
+        reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+        reportWriter.println(TEST_FAILED_MESSAGE);
+      } else {
+        reportWriter.println(TEST_PASSED_MESSAGE);
+      }
+    }
+
+    return canFindClass ? (canFindFS ? Status.SUCCESS : Status.FAIL_TO_FIND_FS)
+        : Status.FAIL_TO_FIND_CLASS;
+  }
+
+  /**
+   * Cleans the HDFS input and output files.
+   */
+  private void cleanFileSystem() throws Exception {
+    if (mFileSystem.exists(mInputFilePath)) {
+      mFileSystem.delete(mInputFilePath, true);
+    }
+    if (mFileSystem.exists(mOutputFilePath)) {
+      mFileSystem.delete(mOutputFilePath, true);
+    }
+    mFileSystem.close();
+  }
+
+  /**
+   * Implements MapReduce with Alluxio integration checker.
+   *
+   * @return 0 for success, 2 for unable to find Alluxio classes, 1 otherwise
+   */
+  private int run(String[] args) throws Exception {
+    Configuration conf = new Configuration();
+    String[] otherArgs =  new GenericOptionsParser(conf, args).getRemainingArgs();
+
+    createHdfsFilesystem(conf);
+    int inputSplits = Integer.parseInt(otherArgs[0]);
+    createInputFile(inputSplits);
+
+    Job job = Job.getInstance(conf, "MapReduceIntegrationChecker");
+    job.setJarByClass(MapReduceIntegrationChecker.class);
+    job.setMapperClass(CheckStatusMapper.class);
+    job.setCombinerClass(MergeStatusReducer.class);
+    job.setReducerClass(MergeStatusReducer.class);
+    job.setOutputKeyClass(Text.class);
+    job.setOutputValueClass(Text.class);
+
+    FileInputFormat.addInputPath(job, mInputFilePath);
+    FileOutputFormat.setOutputPath(job, mOutputFilePath);
+
+    if (!job.waitForCompletion(true)) {
+      cleanFileSystem();
+      return 1;
+    }
+
+    Status resultStatus = generateReport();
+    cleanFileSystem();
```

Comment:
we could use `try ... finally` to make sure `cleanFileSystem` gets called, even when an error is thrown or we return early